src/topic/TopicMessageQuery.js

```diff
@@ -191,83 +322,109 @@ export default class TopicMessageQuery {
-                        /** @type {proto.IConsensusMessageChunkInfo} */ (
-                            message.chunkInfo
-                        );
-                    const initialTransactionID =
-                        /** @type {proto.ITransactionID} */ (
-                            chunkInfo.initialTransactionID
-                        );
-                    const total = /** @type {number} */ (chunkInfo.total);
-                    const transactionId =
-                        TransactionId._fromProtobuf(
-                            initialTransactionID
-                        ).toString();
-
-                    /** @type {proto.ConsensusTopicResponse[]} */
-                    let responses = [];
-
-                    const temp = list.get(transactionId);
-                    if (temp == null) {
-                        list.set(transactionId, responses);
-                    } else {
-                        responses = temp;
+                },
+                (error) => {
+                    if (this._handle != null) {
+                        this._handle.unsubscribe();
                     }
 
-                    responses.push(message);
-
-                    if (responses.length === total) {
-                        const topicMessage = TopicMessage._ofMany(responses);
+                    if (
+                        this._attempt < this._maxAttempts &&
+                        this._retryHandler(error)
+                    ) {
+                        this._attempt += 1;
 
-                        list.delete(transactionId);
-
-                        try {
-                            listener(topicMessage);
-                        } catch (error) {
-                            errorHandler(topicMessage, error);
-                        }
+                        setTimeout(() => {
+                            this._makeServerStreamRequest(client);
+                        }, 250 * 2 ** this._attempt);
                     }
                 }
-            });
+            );
+
+        if (this._handle != null) {
+            this._handle._setCall(() => cancel());
+        }
+    }
+
+    requireNotSubscribed() {
+        if (this._handle != null) {
+            throw new Error(
+                "Cannot change fields on an already subscribed query"
+            );
+        }
+    }
 
-        handle._setCall(() => cancel());
+    /**
+     * @private
+     * @param {TopicMessage} topicMessage
+     */
+    _passTopicMessage(topicMessage) {
+        try {
+            this._counter += 1;
+
+            if (this._listener != null) {
+                this._listener(topicMessage);
+            } else {
+                throw new Error("(BUG) listener is unexpectedly not set");
+            }
+
+            if (this._limit != null && this._limit.comp(this._counter)) {
+                this._completionHandler();
+            }
+        } catch (error) {
+            this._errorHandler(topicMessage, error);
```

Comment:
I removed calling error handler in Java when the application onnext throws an error.

Reply:
This was reverted. It's a large breaking change.